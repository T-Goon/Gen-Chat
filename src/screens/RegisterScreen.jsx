import React, { useState, useContext, useCallback, useEffect } from "react";
import { View, Text, StyleSheet } from "react-native";

import FormInput from '../components/FormInput';
import ButtonPrim from '../components/ButtonPrim';
import LoginRegHere from '../components/LoginRegHere';

import { Context } from '../Context';

const RegisterScreen = ({ navigation }) => {
    const { save, getValueFor } = useContext(Context);

    const [username, setUsername] = useState('');
    const [password, setPassword] = useState('');
    const [confirmPassword, setConfirmPassword] = useState('');

    useEffect(() => {
        getValueFor('token')
            .then((token) => {
                if(token) {
                    navigation.replace('Feed');
                }
            })
            .catch((err) => {
                alert('An error has occurred.');
            });
    }, []);

    const register = useCallback(() => {
        fetch('http://localhost:3000/api/register', {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({ username, password, confirmPassword })
        }).then(async (res) => {
            const data = await res.json();

            if(!data.error) {
                save('token', data.token);
                navigation.replace('Feed');
            } else {
                alert(data.error);
            }
        }).catch((err) => {
            alert('An error occured. Failed to register.');
        });
    }, [username, password, confirmPassword]);

    return (
        <View style={styles.container}>
            <Text style={styles.header}>Gen Chat</Text>

            <View style={styles.form}>

                <Text style={styles.title}>Register</Text>

                <FormInput containerStyles={{ paddingHorizontal: 15, marginVertical: 20 }}
                    label='Username:' placeholder='username' onChangeText={setUsername} />
                <FormInput containerStyles={{ paddingHorizontal: 15, marginVertical: 5 }}
                    label='Password:' placeholder='password' secureTextEntry={true} onChangeText={setPassword} />
                <FormInput containerStyles={{ paddingHorizontal: 15, marginVertical: 5, marginBottom: 20 }}
                    label='Confirm Password:' placeholder='password' secureTextEntry={true} onChangeText={setConfirmPassword} />

                <ButtonPrim buttonStyles={styles.loginButton}
                    onPress={register}>Submit</ButtonPrim>
                <LoginRegHere navigation={navigation} msg={'Already have an account? Login '} location='Login' />
            </View>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        marginTop: 24
    },
    header: {
        position: 'absolute',
        top: 0,
        fontSize: 50
    },
    form: {
        width: '80%',
        paddingVertical: 50,
        borderRadius: 30,
        borderColor: 'black',
        borderWidth: 3,
        backgroundColor: 'hsl(230, 100%, 80%)',
        alignSelf: 'center',
        alignItems: 'center',
        justifyContent: 'center',
    },
    title: {
        fontSize: 40,
        fontWeight: 'bold'
    },
    loginButton: {
        alignSelf: 'center',
        marginBottom: 20
    }
});

export default RegisterScreen;