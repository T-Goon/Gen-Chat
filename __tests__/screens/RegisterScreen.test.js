import React from 'react';
// import renderer from 'react-test-renderer';
import { render, fireEvent, act } from '@testing-library/react-native';

import RegisterScreen from '../../src/screens/RegisterScreen';

describe('<RegisterScreen/>', () => {
    it('has i child', async () => {
        const tree = render(<RegisterScreen />).toJSON();

        expect(tree.children.length).toBe(2);
    });

    it('renders correctly', async () => {
        const tree = render(<RegisterScreen />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});