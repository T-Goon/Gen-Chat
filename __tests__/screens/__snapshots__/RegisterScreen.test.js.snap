// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterScreen/> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 24,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 50,
        "position": "absolute",
        "top": 0,
      }
    }
  >
    Gen Chat
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "hsl(230, 100%, 80%)",
        "borderColor": "black",
        "borderRadius": 30,
        "borderWidth": 3,
        "justifyContent": "center",
        "paddingVertical": 50,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 40,
          "fontWeight": "bold",
        }
      }
    >
      Register
    </Text>
    <View
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
          "marginVertical": 20,
          "paddingHorizontal": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={Object {}}
      >
        Username:
      </Text>
      <TextInput
        allowFontScaling={true}
        keyboardType="default"
        multiline={false}
        onChangeText={[Function]}
        placeholder="username"
        rejectResponderTermination={true}
        secureTextEntry={false}
        style={
          Object {
            "backgroundColor": "hsl(0, 0%, 95%)",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 2,
            "paddingHorizontal": 10,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
          "marginVertical": 5,
          "paddingHorizontal": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={Object {}}
      >
        Password:
      </Text>
      <TextInput
        allowFontScaling={true}
        keyboardType="default"
        multiline={false}
        onChangeText={[Function]}
        placeholder="password"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "hsl(0, 0%, 95%)",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 2,
            "paddingHorizontal": 10,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
          "marginBottom": 20,
          "marginVertical": 5,
          "paddingHorizontal": 15,
          "width": "100%",
        }
      }
    >
      <Text
        style={Object {}}
      >
        Confirm Password:
      </Text>
      <TextInput
        allowFontScaling={true}
        keyboardType="default"
        multiline={false}
        onChangeText={[Function]}
        placeholder="password"
        rejectResponderTermination={true}
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "hsl(0, 0%, 95%)",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 2,
            "paddingHorizontal": 10,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "hsl(550, 100%, 50%)",
          "borderRadius": 10,
          "justifyContent": "center",
          "marginBottom": 20,
          "opacity": 1,
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Submit
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text>
        Already have an account? Login 
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          style={
            Object {
              "borderBottomColor": "red",
              "borderBottomWidth": 1,
              "color": "red",
            }
          }
        >
          here
        </Text>
      </View>
    </View>
  </View>
</View>
`;
